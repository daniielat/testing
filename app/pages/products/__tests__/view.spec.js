const React = require('react');
const View = require('../view');
const { render, screen } = require('@testing-library/react');
const mockProducts = require('../../../../mocks/test/get/https/api.mercadolibre.com/sites/MLA/search?limit=10&offset=0&q=celular.json');

describe('La View de Products', () => {
    let component;
    const i18n = { gettext: text => text };
    
    it('Renderiza', () => {
        component = render(<View i18n={i18n} products={mockProducts.data.data}/>)
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();
    });

    it('Renderiza un mensaje de error si no se encontraron productos', () => {
        component = render(<View i18n={i18n} products={[]}/>);
        const message = screen.getByText(/no se encontraron productos/i);
        expect(message).toBeInTheDocument();
    });
});
