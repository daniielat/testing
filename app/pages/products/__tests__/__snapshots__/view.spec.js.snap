// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`La View de Products Renderiza 1`] = `
<DocumentFragment>
  <div
    class="products"
  >
    <section>
      <h2>
        Listado de productos
      </h2>
      <section>
        <button
          class="andes-button andes-button--large andes-button--quiet"
          type="button"
        >
          <span
            class="andes-button__content"
          >
            Siguiente
          </span>
        </button>
      </section>
      <ol>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
              Celular Zte Blade A31 Plus Gris Rva 1/32gb Nuevo Garantía
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt="Celular Zte Blade A31 Plus Gris Rva 1/32gb Nuevo Garantía"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_979817-MLA51002594618_082022-I.jpg"
            />
          </div>
        </div>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
               Moto G22 128 Gb  Cosmic Black 4 Gb Ram
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt=" Moto G22 128 Gb  Cosmic Black 4 Gb Ram"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_854793-MLA50262132804_062022-I.jpg"
            />
          </div>
        </div>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
              Samsung Galaxy A32 128 Gb  Awesome Blue 4 Gb Ram
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt="Samsung Galaxy A32 128 Gb  Awesome Blue 4 Gb Ram"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_878776-MLA48428092348_122021-I.jpg"
            />
          </div>
        </div>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
              Samsung Galaxy A03 128 Gb  Negro 4 Gb Ram
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt="Samsung Galaxy A03 128 Gb  Negro 4 Gb Ram"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_737913-MLA49433851234_032022-I.jpg"
            />
          </div>
        </div>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
              Celular Samsung Galaxy A13 128gb + 4gb Ram Pls Lcd Negro
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt="Celular Samsung Galaxy A13 128gb + 4gb Ram Pls Lcd Negro"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_917545-MLA50401654103_062022-I.jpg"
            />
          </div>
        </div>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
              Samsung Galaxy A03 Core 32 Gb  Black 2 Gb Ram
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt="Samsung Galaxy A03 Core 32 Gb  Black 2 Gb Ram"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_604790-MLA49152794859_022022-I.jpg"
            />
          </div>
        </div>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
               Moto G60s 128 Gb  Azul 6 Gb Ram
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt=" Moto G60s 128 Gb  Azul 6 Gb Ram"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_788340-MLA48112659213_112021-I.jpg"
            />
          </div>
        </div>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
              Samsung Galaxy A13 64 Gb  White 4 Gb Ram
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt="Samsung Galaxy A13 64 Gb  White 4 Gb Ram"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_762562-MLA50401440917_062022-I.jpg"
            />
          </div>
        </div>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
              Samsung Galaxy A03 64 Gb  Negro 4 Gb Ram
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt="Samsung Galaxy A03 64 Gb  Negro 4 Gb Ram"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_737913-MLA49433851234_032022-I.jpg"
            />
          </div>
        </div>
        <div
          class="andes-card andes-card--flat andes-card--default andes-card--padding-default"
        >
          <div
            class="andes-card__header andes-card__header--border"
          >
            <h2
              class="andes-card__title"
            >
              Motorola G51 5g 128 Gb  Dorado 4 Gb Ram
            </h2>
          </div>
          <div
            class="andes-card__content"
          >
            <img
              alt="Motorola G51 5g 128 Gb  Dorado 4 Gb Ram"
              class=""
              decoding="async"
              src="http://http2.mlstatic.com/D_959500-MLA50302263673_062022-I.jpg"
            />
          </div>
        </div>
      </ol>
    </section>
  </div>
</DocumentFragment>
`;
